dcapp/app/src/main/java/com/example/michael/mqtttest/BroadcastService.java

```diff
@@ -0,0 +1,239 @@
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
+import android.os.IBinder;
+import android.support.annotation.Nullable;
+
+import org.eclipse.paho.android.service.MqttAndroidClient;
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * @desc - This is the work horse for acceleration data broadcasting. When
+ *         this service is run,it will attempt to connect to an MQTT broker at
+ *         the IP address provided by the activity calling this service. The
+ *         AccelerationListener class in this service is responsible for
+ *         calling the 'publish' method of the MQTT client to send data to the
+ *         MQTT broker. The acceleration listener is also responsible for
+ *         detecting when the device is disconnected and perpetually
+ *         attempting to re-establish connection.
+ *
+ *         TO-DO:
+ *         We should consider moving the responsibility of publishing data
+ *         from the AccelerationListener to a running thread that triggers a
+ *         publish event in discrete time and have the AccelerationListener
+ *         change the values published rather than tirggering publications on
+ *         acceleration events based on a delay
+ */
+public class BroadcastService extends Service {
+    /* Use this to determine if the service is running */
+    public static Boolean broadcastServiceIsRunning = false;
+
+    /* Use this to show the connection status */
+    public static String connectionStatus = "Disconnected";
+
+    /* Use this to show the publish rate */
+    public static long publishRateMilliSec = 0;
+
+
+    // Use this to enable delays
+    private int tickAccumulator = 0;
+    private long lastTimeCheck = System.currentTimeMillis();
+    private MqttAndroidClient client;
+    private int delayValue = 0;
+    private String androidId = "NOTHING";
+    private SensorManager sensorManager;
+    private Sensor accelerometer;
+    private String hostIp;
+    SensorEventListener accelerationListener;
+
+    // More constants (probably should store in a class)
+    private static final String TCP_PREFIX = "tcp://";
+
+    private static final String DEVICE_ID_INTENT_EXTRA = "AndroidId";
+    private static final String HOST_IP_INTENT_EXTRA = "hostIp";
+    private static final String SPEED_SETTING_INTENT_EXTRA = "userSpeedInput";
+
+    private static final String NO_IP_ADDRESS_FOUND = "NO ADDRESS FOUND";
+    private static final String NO_ANDROID_ID_FOUND = "NO DATA";
+    private static final int NO_USER_SPEED_INPUT = 0;
+
+    private static final String WATCH_ID_JSON_INDEX = "watch_id";
+    private static final String TIMESTAMP_JSON_INDEX = "timestamp";
+    private static final String ACC_X_JSON_INDEX = "acc_x";
+    private static final String ACC_Y_JSON_INDEX = "acc_y";
+    private static final String ACC_Z_JSON_INDEX = "acc_z";
+
+    // This should change
```

Comment:
A lot of those constants are shared in the MainActivity class. Didn't bother creating a constants class but think it would be a good idea.